components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/APIClient.java

```diff
@@ -239,6 +269,42 @@ private <T> T deserialize(HttpEntity responseEntity, Class<T> returnType) throws
         }
     }
 
+    private String validateJsonResponse(HttpEntity responseEntity) throws ActionInvocationException {
+
+        if (!isAcceptablePayload(responseEntity)) {
+            throw new ActionInvocationException("The response content type is not application/json.");
+        }
+
+        try {
+            return EntityUtils.toString(responseEntity);
+        } catch (IOException e) {
+            throw new ActionInvocationException("Parsing JSON response failed.", e);
+        }
+    }
+
+    private ActionInvocationResponse.APIResponse deserialize(HttpEntity responseEntity)
+            throws ActionInvocationException {
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            String jsonResponse = validateJsonResponse(responseEntity);
+            JsonNode rootNode = objectMapper.readTree(jsonResponse);
+            String actionStatus = rootNode.path(ACTION_STATUS).asText();
```

Comment:
what happens if rootNode does not include the path "actionStatus" ?

Reply:
This has been updated with <WDSDSDSDASDASDSd> to throw an ActionInvocationResponseException if such a property is unavailable. Currently, this scenario is being treated as a violation of the defined API contract.